### ISSUE TITLE
Typescript is complaining when the image argument is "string | null"

### ISSUE THREAD

**Issue by PSalaun** [bundlesandbatches/next-sanity-image#13]
Hello,

I'm creating a component that may or may not receive an image from Sanity, so the image type is "string | null". However TS is not happy with it (using version 3.1.4), as I get `Argument of type 'string | null' is not assignable to parameter of type 'null'. Type 'string' is not assignable to type 'null'.` when passing the image to `useNextSanityImage`. It does the same if I replace string by SanityImageSource.

As the hook can handle null images, it should be ok to pass this argument, shouldn't it?

**Comment by lorenzodejong** [comment]
Thanks for reporting this bug! This was a recently introduced feature and it looks like this only occurs when explicitly typing the image param. I've just released 3.1.5 with a fix, please let me know if this solves your issue!

**Comment by 3200pro** [comment]
Hi,

I'm running Next.js V12 and just ran into this problem as well:

```
export const urlFor = (source: any) =>
  createImageUrlBuilder(config).image(source);

export function useSanityImage(image: SanityImageSource | null) {
  const data = useNextSanityImage(getClient(false), image);

  return data!;
}
```

Error:

```
./lib/sanity.ts:21:53
Type error: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'SanityImageSource | null' is not assignable to parameter of type 'SanityImageSource'.
      Type 'null' is not assignable to type 'SanityImageSource'.
```

Let me know if I'm missing something or if there's an error on my end.
